frontend/src/components/elements/Progress/Progress.tsx

```diff
@@ -15,69 +15,25 @@
  * limitations under the License.
  */
 
-import React, { PureComponent, ReactNode } from "react"
+import React, { ReactElement } from "react"
 import { Map as ImmutableMap } from "immutable"
-import { Progress as UIProgress } from "reactstrap"
+import ProgressBar from "components/shared/ProgressBar"
 
-import "./Progress.scss"
-
-export interface Props {
+export interface ProgressProps {
   width: number
   element: ImmutableMap<string, any>
 }
 
 export const FAST_UPDATE_MS = 50
 
-class Progress extends PureComponent<Props> {
-  lastValue = -1
-
-  lastAnimatedTime = -1
-
-  isMovingBackwards = (): boolean => {
```

Comment:
There's a lot of logic associated with this as a result of https://github.com/streamlit/streamlit/commit/1f3dc8fe3110343b6b83866b9cc0c8510e728e23 Going to talk with @arraydude to see what cases I should consider.

Reply:
So I tested this out, and it works if it's at 100% from https://github.com/streamlit/streamlit-old-private/issues/588 It does do a little reanimate in the middle. I am not sure if this is a big issue.